### DISCUSSION TITLE
queryRaw not using ORDER BY clause with postgres

### DISCUSSION THREAD

**Discussion by ajoslin103** [prisma/prisma#10231] [Q&A] [answered]
I am using prisma 3.3.0 over postgres 

My ORDER BY clause is not being respected in the $queryRaw results on my materialized view  (created using records from 3 tables)

If I copy the query from the log and execute it in TablePlus the ORDER BY clause is respected.

I have even hardcoded the statement, and still the ORDER BY clause is ignored

```ts
await db.$queryRaw`SELECT * FROM "CitizenView" WHERE "eventId" = 'O-R9i1Wu8obbWgYm6hgUC' ORDER BY name DESC LIMIT 5 OFFSET 5`
```
I do not know how to solve this problem

Thanks for any help you can offer!

Al;

I create the materialized view thusly:

```sql
DROP MATERIALIZED VIEW "CitizenView";

CREATE MATERIALIZED VIEW "CitizenView"
AS
SELECT id, "updated_at", "name", "phone", "eventId", "answerStatus", 'Attendee' AS "citizenType", "hasGuests" ,CASE WHEN "twoFACode" IS NOT NULL = TRUE then TRUE ELSE FALSE END AS "hasTwoFA" FROM "Attendee"
UNION
SELECT id, "updated_at", "name", "phone", "eventId", "answerStatus", 'Forward' AS "citizenType", FALSE AS "hasGuests", CASE WHEN "twoFACode" IS NOT NULL = TRUE then TRUE ELSE FALSE END AS "hasTwoFA" FROM "Forward"
UNION
SELECT id, "updated_at", "name", "phone", "eventId", "answerStatus", 'eventId' AS "citizenType", FALSE AS "hasGuests", FALSE AS "hasTwoFA" FROM "Guest"
WITH NO DATA;

REFRESH MATERIALIZED VIEW "CitizenView";
```

I query against the materialized view thusly:

```ts
logger.debug( `[${__file}] ~ SELECT * FROM "CitizenView" WHERE "eventId" = ${eventId} ORDER BY ${orderByStr} LIMIT ${pageSize} OFFSET ${rowOffset}` ) // note: prisma needs to do all interpolation in order to prepare the statement
rows = await db.$queryRaw`SELECT * FROM "CitizenView" WHERE "eventId" = ${eventId} ORDER BY ${orderByStr} LIMIT ${pageSize} OFFSET ${rowOffset}`
logger.debug( `[${__file}] ~ SORTED rows: ${JSON.stringify(rows, null, 2)}` )
```

I only attempt to sort against the "name" and "phone" columns

Nothing I try gives me sorted results

**Comment by ajoslin103** [comment, marked as answer]
I switched to `queryRawUnsafe` and it works now

Unless I hear otherwise in a few days I might switch this over to a bug.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
